bus-mapping/src/evm/opcodes/callop.rs

```diff
@@ -632,13 +632,15 @@ fn write_memory_words(
 
     // Add memory word write ops.
     for (i, chunk) in slot_bytes.chunks(32).enumerate() {
+        let chunk_prev = chunk; // TODO: get previous value
```

Comment:
yes, the `chunk_prev` is not real chunk before the writing,  one solution I can think out is reading  from `let prev_memory = state.call_ctx_mut()?.memory ` before Line#186 , here read prev chunk by prev_memory

Reply:
Correct.

I also experimented doing it in a common function `CircuitInputStateRef::memory_word_write`

Reply:
Now everything is done with the function `memory_word_write`.